pkg/apis/psmdb/v1/psmdb_types.go

```diff
@@ -122,9 +123,18 @@ func (us UpgradeStrategy) Lower() UpgradeStrategy {
 	return UpgradeStrategy(strings.ToLower(string(us)))
 }
 
+func OneOfUpgradeStrategy(a string) bool {
+	return strings.ToLower(a) == string(UpgradeStrategyLatest) ||
+		strings.ToLower(a) == string(UpgradeStrategyRecommended) ||
+		strings.ToLower(a) == string(UpgradeStrategyDiasbled) ||
+		strings.ToLower(a) == string(UpgradeStrategyNever)
```

Comment:
i think it's worth to accept `UpgradeStrategy` instead of casting all constants

Reply:
In any case we can pass any string here. So we should lowercse it which requires casting to string.
So I simplify it a bit, but accept string https://github.com/percona/percona-server-mongodb-operator/pull/625/commits/a7211780084d45514a1eb75fdf8f0ca2f2bbf0fb